### PR TITLE
Create remote-replay-launcher package & command

### PR DESCRIPTION
**Pull request by alexivanov**
### Changes

This PR creates a package(`remote-replay-launcher`) with a simple API for execution of a remote test run (a cloud replay test run against a local target) and a CLI command.

The report-diiffs-action will call into the `remote-replay-launcher` package directly.

### Code review

Commits are logically split so review by commit might be easier.

### CODE REVIEW

File: packages/cli/src/commands/remote-run-all-tests/remote-run-all-tests.command.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    progressBar.update(progressBar.getTotal());
+
+    progressBar.stop();
+  }
+};
+
+export const remoteRunAllTestsCommand = buildCommand("remote-run-all-tests")
+  .details({ describe: "Run all replay test cases remotely" })
+  .options({
+    apiToken: OPTIONS.apiToken,
```
Comment (Que3216) [inline packages/cli/src/commands/remote-run-all-tests/remote-run-all-tests.command.ts:97]: Unrelated to this PR but I wonder if we should make apiToken required if process.env["METICULOUS_API_TOKEN"] not set?

File: packages/cli/src/commands/remote-run-all-tests/remote-run-all-tests.command.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      throw error;
+    }
+  } finally {
+    progressBar.update(progressBar.getTotal());
+
+    progressBar.stop();
+  }
+};
+
+export const remoteRunAllTestsCommand = buildCommand("remote-run-all-tests")
```
Comment (Que3216) [inline packages/cli/src/commands/remote-run-all-tests/remote-run-all-tests.command.ts:94]: Other name options:
 - `run-all-tests-remote`
 - `run-all-tests-remotely`
 - `run-all-tests-in-cloud`

I think I like `run-all-tests-remotely` or `run-all-tests-in-cloud` the most.

Comment (Que3216) [reply packages/cli/src/commands/remote-run-all-tests/remote-run-all-tests.command.ts:94]: Maybe `run-all-tests-in-cloud`?

**Review by Que3216 — APPROVED** [review]
Nice!

File: packages/cli/src/commands/remote-run-all-tests/remote-run-all-tests.command.ts (hunk 2 above)
Comment (alexivanov) [reply packages/cli/src/commands/remote-run-all-tests/remote-run-all-tests.command.ts:94]: 👍 let's go with `run-all-tests-in-cloud`

Comment (alexivanov) [reply packages/cli/src/commands/remote-run-all-tests/remote-run-all-tests.command.ts]: https://github.com/alwaysmeticulous/meticulous-sdk/pull/595/commits/b70ce483bdce4520a0a378e850b549aec443c78a

**Pull request merged**